examples/asr/speech_to_text.py

```diff
@@ -62,16 +55,86 @@
     +model.optim.args.betas=[0.8,0.5]\
     +model.optim.args.weight_decay=0.0005
 
+Finetune a model
+1) Finetune from a .nemo file
+
+```sh
+    python examples/asr/speech_to_text.py \
+        --config-path=<path to dir of configs> \
+        --config-name=<name of config without .yaml>) \
+        model.train_ds.manifest_filepath="<path to manifest file>" \
+        model.validation_ds.manifest_filepath="<path to manifest file>" \
+        trainer.gpus=-1 \
+        trainer.max_epochs=50 \
+        +init_from_nemo_model="<path to .nemo model file>"
+```
+
+2) Finetune from a pretrained model (via NGC)
+
+```sh
+    python examples/asr/speech_to_text.py \
+        --config-path=<path to dir of configs> \
+        --config-name=<name of config without .yaml>) \
+        model.train_ds.manifest_filepath="<path to manifest file>" \
+        model.validation_ds.manifest_filepath="<path to manifest file>" \
+        trainer.gpus=-1 \
+        trainer.max_epochs=50 \
+        +init_from_pretrained_model="<name of pretrained checkpoint>"        
+```
 """
 
+import pytorch_lightning as pl
+from omegaconf import OmegaConf, open_dict
+
+from nemo.collections.asr.models import EncDecCTCModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+
+
+def restore_weights_if_required(model: EncDecCTCModel, cfg: OmegaConf):
+    if 'init_from_nemo_model' not in cfg and 'init_from_pretrained_model' not in cfg:
+        # model weights do not need to be restored
+        return
+
+    if 'init_from_nemo_model' in cfg and 'init_from_pretrained_model' in cfg:
+        raise ValueError("Cannot pass both `init_from_nemo_model` and `init_from_pretrained_model` to config!")
+
+    if 'init_from_nemo_model' in cfg and cfg.init_from_nemo_model is not None:
+        with open_dict(cfg):
+            # Restore model
+            model_path = cfg.pop('init_from_nemo_model')
+            restored_model = EncDecCTCModel.restore_from(model_path, map_location='cpu', strict=True)
+
+            # Restore checkpoint into current model
+            model.load_state_dict(restored_model.state_dict(), strict=False)
+            logging.info(f'Model checkpoint restored from nemo file with path : `{model_path}`')
+
+            del restored_model
+
+    if 'init_from_pretrained_model' in cfg and cfg.init_from_pretrained_model is not None:
+        with open_dict(cfg):
+            # Restore model
+            model_name = cfg.pop('init_from_pretrained_model')
+            restored_model = EncDecCTCModel.from_pretrained(model_name, map_location='cpu', strict=True)
+
+            # Restore checkpoint into current model
+            model.load_state_dict(restored_model.state_dict(), strict=False)
```

Comment:
If we just need the state of the model, can we add the feature to support PTL checkpoints too?
It happens a lot that the training is not done and we need to download the checkpoint, copy the tokenizer to our local system with the same exact location in order to convert a checkpoint to nemo file.